influxql/select.go

```diff
@@ -355,6 +354,25 @@ func buildRHSTransformIterator(lhs Iterator, rhs Literal, op Token, ic IteratorC
 				return p
 			},
 		}, nil
+	case func(int64, int64) int64:
```

Comment:
This kind of stuff is going to have to be refactored to use the casting I mentioned before.

One of the conflicts we have for transform iterators is that number literals have no way of identifying a float or an integer, so we always have to assume float. So the only time you can have an integer to integer transform is when there are two iterators with a binary expression operator. Any other time and you will have to perform a float conversion for the integer iterator and output a float. We can't tell the difference right now between `2.0` and `2`. This code here won't operate correctly if you have an integer field and do `value * 2.5`.